src/shaders/pbr.frag

```diff
@@ -137,96 +149,99 @@ float geometrySchlickGGX(float dotProd, float roughness) {
+void microfacetModel(vec3 specularReflectance,
+                     vec3 c_diff,
                      float metallic,
                      float roughness,
-                     vec3 normal,
-                     vec3 light,
-                     vec3 view,
-                     vec3 lightRadiance) {
-  vec3 halfVector = normalize(light + view);
-  // compute F0, specular reflectance at normal incidence
-  // for nonmetal, using constant 0.04
-  vec3 F0 = mix(vec3(0.04), baseColor, metallic);
-  vec3 Fresnel = fresnelSchlick(F0, view, halfVector);
-
+                     float v_dot_h,
+                     float n_dot_l,
+                     float n_dot_v,
+                     float n_dot_h,
+                     vec3 lightRadiance,
+                     out vec3 diffuseContrib,
+                     out vec3 specularContrib) {
+  vec3 Fresnel = fresnelSchlick(specularReflectance, v_dot_h);
   // Diffuse BRDF
   // NOTE: energy conservation requires
   // diffuse + specular <= 1.0, where specular = Fresnel
-  // Also: result does not need to be scaled by 1/PI
-  // See details:
-  // https://seblagarde.wordpress.com/2012/01/08/pi-or-not-to-pi-in-game-lighting-equation/
-  vec3 diffuse = mix(vec3(1.0) - Fresnel, vec3(0.0), metallic) * baseColor;
-
-  // Specular BDDF
-#if defined(DOUBLE_SIDED)
-  float temp =
-      max(4.0 * abs(dot(normal, light)) * abs(dot(normal, view)), Epsilon);
-#else
-  float temp =
-      max(4.0 * max(dot(normal, light), 0.0) * max(dot(normal, view), 0.0),
-          Epsilon);
-#endif
-  vec3 specular = Fresnel *
-                  specularGeometricAttenuation(normal, light, view, roughness) *
-                  normalDistribution(normal, halfVector, roughness) / temp;
+  vec3 diffuse = (vec3(1.0) - Fresnel) * c_diff * INV_PI;
 
-  return (diffuse + specular) * lightRadiance *
-#if defined(DOUBLE_SIDED)
-  abs(dot(normal, light));
-#else
-  max(dot(normal, light), 0.0);
-#endif
+  // Specular BRDF
+  float temp = max(4.0 * n_dot_l * n_dot_v, Epsilon);
+  vec3 specular = Fresnel *
+                  specularGeometricAttenuation(n_dot_l, n_dot_v, roughness) *
+  // normalDistributionGGX is defined in the pbrCommon.glsl
+  // Specular D, normal distribution function (NDF),
+  // also known as ggxDistribution
+                  normalDistributionGGX(n_dot_h, roughness) / temp;
+
+  vec3 tempVec = lightRadiance * n_dot_l;
+  diffuseContrib = diffuse * tempVec;
+  specularContrib = specular * tempVec;
 }
 
 void main() {
   vec3 emissiveColor = Material.emissiveColor;
 #if defined(EMISSIVE_TEXTURE)
-  emissiveColor *= texture(EmissiveTexture, texCoord).rgb;
+  emissiveColor *= SRGBtoLINEAR(texture(EmissiveTexture, texCoord)).rgb;
 #endif
   fragmentColor = vec4(emissiveColor, 0.0);
 
 #if (LIGHT_COUNT > 0)
   vec4 baseColor = Material.baseColor;
 #if defined(BASECOLOR_TEXTURE)
-  baseColor *= texture(BaseColorTexture, texCoord);
+  baseColor *= SRGBtoLINEAR(texture(BaseColorTexture, texCoord));
```

Comment:
Do we expect to see an overall change in brightness for existing PBR users/scenes? If so, is there anyone we should warn?

For context, I'm thinking about the switch from GenericDrawable to PbrDrawable which caused YCB and other objects to unexpectedly darken for many users.